src/main/java/com/enterprise/fundraisermanager/FundraiserManagerController.java

```diff
@@ -79,6 +79,20 @@ public ResponseEntity deleteFundraiser(@PathVariable("id") Integer id) {
         }
     }
 
+    @PutMapping("/fundraiser/{id}/")
+    public ResponseEntity deleteFundraiserFormUI(@PathVariable("id") Integer id) {
+        try {
+            Fundraiser fundraiser = fundraiserService.fetchById(id);
+            fundraiser.setIsActive(false);
+            fundraiserService.save(fundraiser);
+            return new ResponseEntity(HttpStatus.OK);
+        } catch (Exception e) {
+            //TODO confirm correct response in catch, or modify it to be correct response.
+            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+

```

Comment:
For the most part hard deleting data from the database is not always a good idea. So I added an IsActive field to the fundraiser dto. The idea is that when IsActive is set to false the fundraiser is not shown in the ui. This code in particular is the mapping to set the fundraiser IsActive field to false.